Improve development documentation (#2699)
redhat-appstudio/infra-deployments: docs/development/deployment.md

Note running these scripts in a clone repo will have no effect as the repo will remain `https://github.com/redhat-appstudio/infra-deployments.git`

### `preview.env` instructions

> I want a development deployment of StoneSoup where I can:

Author: Wasn't sure if I should call it StoneSoup, AppStudio or RHTAP. All 3 are used in some places here, StoneSoup and RHTAP with about equal frequency, AppStudio not so much

Reviewer: The latest name is RHTAP but this documentation uses Stonesoup so for consistency within this document, I would use Stonesoup and we should do a separate PR if we want to rename Stonesoup to RHTAP

| `MY_GIT_FORK_REMOTE`          | [Fork the repo](#fork-the-repo)   |
| `MY_GITHUB_ORG`               | [GitHub setup](#github-setup)     |
| `MY_GITHUB_TOKEN`             | [GitHub setup](#github-setup)     |
| `IMAGE_CONTROLLER_QUAY_ORG`   | [Quay setup](#quay-setup)         |
| `IMAGE_CONTROLLER_QUAY_TOKEN` | [Quay setup](#quay-setup)         |
| `PAC_GITHUB_APP_PRIVATE_KEY`  | [PaC setup](#pac-setup)           |
| `PAC_GITHUB_APP_ID`           | [PaC setup](#pac-setup)           |

#### Fork the repo

If you haven't done so yet, fork <https://github.com/redhat-appstudio/infra-deployments> and clone
it locally. Copy [`hack/preview-template.env`][preview-template.env] to `hack/preview.env` - that's
where you want to set all the environment variables.

The first one is `MY_GIT_FORK_REMOTE` - the name of the git remote which points to your fork.

```shell
$ git remote -v
origin  git@github.com:<gh_username>/infra-deployments.git (fetch)
origin  git@github.com:<gh_username>/infra-deployments.git (push)
upstream    https://github.com/redhat-appstudio/infra-deployments.git (fetch)
upstream    https://github.com/redhat-appstudio/infra-deployments.git (push)
```

```shell

> export MY_GIT_FORK_REMOTE=origin

Reviewer 2: How does this work?

Reviewer 2: Ah, I see..it's the actual line that's in the preview.env file. Confused me a bit :D So maybe I would make it even more explicit here?

Author: Good point, I'll try to make it more obvious

copy its value to `preview.env`:

```shell
export IMAGE_CONTROLLER_QUAY_TOKEN=***
```

#### PaC setup

Follow the [Pipelines as Code: Setup Manually][pac-setup-manual] instructions to create the GitHub application.
You can create the application in your organization settings (<https://github.com/organizations/MY_STONESOUP_ORG/settings/apps>)
or your personal settings (<https://github.com/settings/apps>). If you create the app under your organization,
you can allow it `Only on this account`, otherwise you will have to allow `Any account`.

Compared to the Pipelines as Code documentation, you will want to set these fields differently:

* **GitHub Application Name**: use a unique name, e.g. `<gh_username>-stonesoup PaC`
* **Webhook URL**: use a dummy url, e.g. <https://example.org>. The `preview.sh` script will fix the
  webhook url automatically.
* **Webhook secret**: leave it empty, the `preview.sh` script will set the secret automatically.

Once you have created the GitHub app, copy its App ID to `preview.env`:

```shell
export PAC_GITHUB_APP_ID=<number>
```

Generate a private key for the application (in the GitHub UI; refer to the Pipelines as Code docs linked

> above). Encode the private key as base64 and copy it to `preview.env`:
> 
> ```shell
> base64 --wrap=0 <path to downloaded key>
> ```
> 
> ```shell
> export PAC_GITHUB_APP_PRIVATE_KEY=<output of the command above>
> ```

Reviewer 3: ```suggestion
above). Paste the base64 encoded private key to `preview.env`:
```shell
export PAC_GITHUB_APP_PRIVATE_KEY=$(base64 < /PATH/TO/YOUR/DOWNLOADED/PRIVATE_KEY.pem)
```

something like this? since it's easier

Author: My workflow was copy the base64 encoded key into preview.env and then delete it. Though I guess there's no real reason not to keep the key around :thinking:

Reviewer 3: That's a good point :+1:
